lib/utilities/monitoringHandler.js

```diff
@@ -0,0 +1,65 @@
+const errors = require('arsenal');
+const client = require('prom-client');
+
+function writeResponse(res, error, log, results, cb) {
+    let statusCode = 200;
+    if (error) {
+        if (Number.isInteger(error.code)) {
+            statusCode = error.code;
+        } else {
+            statusCode = 500;
+        }
+    }
+    res.writeHead(statusCode, { 'Content-Type': 'application/json' });
+    res.write(JSON.stringify(results));
+    res.end(() => {
+        cb(error, results);
+    });
+}
+
+
+function routeHandler(req, res, log, cb) {
+    if (req.method !== 'GET') {
+        return cb(errors.BadRequest, []);
+    }
+    client.collectDefaultMetrics();
+    const promMetrics = client.register.metrics();
+    const contentLen = Buffer.byteLength(promMetrics, 'utf8');
+    res.setHeader('content-length', contentLen);
```

Comment:
Shouldn't be required, and maybe even set by the HTTP server module? Anyway, calculating the length can be somewhat expensive without a need.

Reply:
This was suggested by the lead. It is required.